app/assets/javascripts/visualizations/highvis/map.coffee

```diff
@@ -589,26 +590,25 @@ $ ->
 
       clip: (arr) ->
         viewBounds = @gmap.getBounds()
-        if viewBounds?
-          filterFunc = (row) ->
-            lat = lng = null
+        unless viewBounds? then return arr
 
-            # Scan for lat and long
-            for field, fieldIndex in data.fields
-              if (Number field.typeID) in data.types.LOCATION
-                if (Number field.typeID) is data.units.LOCATION.LATITUDE
-                  lat = row[fieldIndex]
-                else if (Number field.typeID) is data.units.LOCATION.LONGITUDE
-                  lng = row[fieldIndex]
+        filterFunc = (row) ->
+          lat = lng = null
 
-            # If points are valid, check if they are visible
-            if (lat is null) or (lng is null)
-              return false
-            else return viewBounds.contains(new google.maps.LatLng(lat, lng))
+          # Scan for lat and long
+          for field, fieldIndex in data.fields
+            if (Number field.typeID) in data.types.LOCATION
+              if (Number field.typeID) is data.units.LOCATION.LATITUDE
+                lat = row[fieldIndex]
+              else if (Number field.typeID) is data.units.LOCATION.LONGITUDE
```

Comment:
If it's in location and not latitude, doesn't that mean it must be longitude?  Can this just be a hard else?